packages/mongo/oplog_observe_driver.js

```diff
@@ -565,20 +558,33 @@ Object.assign(OplogObserveDriver.prototype, {
                   // _pollQuery call (eg, in another fiber) which should
                   // effectively cancel this round of fetches.  (_pollQuery
                   // increments the generation.)
-
-                  self._handleDoc(id, doc);
+                  try {
+                    self._handleDoc(id, doc);
+                    resolve();
+                  } catch (err) {
+                    reject(err);
+                  }
+                } else {
+                  resolve();
                 }
-              } finally {
-                waiting--;
-                // Because fetch() never calls its callback synchronously,
-                // this is safe (ie, we won't call fut.return() before the
-                // forEach is done).
-                if (waiting === 0) promiseResolver();
-              }
-            })
-          );
+              })
+            )
+          })
+          fetchPromises.push(fetchPromise);
         });
-        await awaitablePromise;
+        // Wait for all fetch operations to complete
+        try {
+          const results = await Promise.allSettled(fetchPromises);
+          const errors = results
+            .filter(result => result.status === 'rejected')
+            .map(result => result.reason);
+
+          if (errors.length > 0) {
+            Meteor._debug('Some fetch queries failed:', errors);
+          }
+        } catch (err) {
+          Meteor._debug('Got an exception in a fetch query', err);
+        }
```

Comment:
If I'm not mistaken, all rejections will already be caught in the `allSettled`; we can safely remove this catch block as our errors are being caught in the `errors.length > 0` branch